### ISSUE TITLE
Remove virtual inheritance

### ISSUE THREAD

**Issue by Flamefire** [Return-To-The-Roots/s25client#192]
Virtual inheritance is hard to handle and easy to make mistakes. All usages of virtual inheritance should be re-evaluated and removed where possible.
Example: libsiedler2::ArchivItemBitmap and related

**Comment by Flow86** [comment]
how do you want to remove it, since you have some kind of diamond formed class tree

**Comment by Flamefire** [comment]
And that form should be avoided if possible. Most often this can be solved by introducing another class. And e.g. the bitmap classes might not need a diamond shaped tree here. But got to check that more deeply. I just found that while refactoring libsiedler2 to fix the mem leaks permanently.

**Comment by Flow86** [comment]
yes I know, I hate the shape too, but when I wrote the code, I was unable to find a better solution, but perhaps you have a good idea solving the problem. (perhaps by using some template factory stuff etc)

**Comment by Flamefire** [comment]
Yep. And VS has a nice class inheritance visualizer. Gonna check that once I'm done with the current changes.(They are already huge, but greatly improve the code) But got some minor issues left.
